### PR TITLE
Rename `ipython_mode` to `numbered_prompt`

### PR DESCRIPTION
**Pull request by hyrodium**
I love the feature of `ipython_mode`, but I disagree with its name because:

* The feature is less related to IPython.
* I feel like there's not enough respect for our ancestors who developed Mathematica.

Fortunately, it's before the release of Julia v1.9, and I think it's a good opportunity to reconsider the name.
https://github.com/JuliaLang/julia/pull/46474#issuecomment-1426759419

### CODE REVIEW

**Comment by hyrodium** [comment]
Reactions to my proposal for the renaming seem positive on Slack.

image

https://julialang.slack.com/archives/C67TK21LJ/p1681139172800279

**Comment by KristofferC** [comment]
If IPython mode is too strong, then softening it to e.h. `ipython_prompt` feels ok (since this is clearly not emulating the full IPyhton REPL experience). The suggested `numbered_mode´ feels very bland.

**Comment by pabloferz** [comment]
Why not `notebook_mode`?

**Comment by KristofferC** [comment]
Because that makes it sounds like it provides some notebook functionality. As I said before, I do think the `mode` term is a bit too broad. `notebook_prompt` might be ok.

**Comment by IanButterworth** [comment]
`numbered_prompt` seems good

**Comment by hyrodium** [comment]
I think `numbered_prompt`  is the current best choice, so I have updated this PR.

**Comment by JeffBezanson** [comment]
:100: This is much better; the current name makes it sound like there is some kind of full IPython emulation, or that it will actually run IPython, or something. And IPython is not even called that anymore :wink:

File: HISTORY.md [hunk 1]
```diff
@@ -135,8 +135,8 @@ Standard library changes
 * The contextual module which is active in the REPL can be changed (it is `Main` by default),
   via the `REPL.activate(::Module)` function or via typing the module in the REPL and pressing
   the keybinding Alt-m ([#33872]).
-* An "IPython mode" which mimics the behaviour of the prompts and storing the evaluated result in `Out` can be
-  activated with `REPL.ipython_mode!()`. See the manual for how to enable this at startup ([#46474]).
+* An "Numbered prompt" which mimics the behaviour of the prompts and storing the evaluated result in `Out` can be
```
Comment (JeffBezanson) [inline HISTORY.md:138]:
```suggestion
* A "numbered prompt" mode which prints numbers for each input and output and stores evaluated results in `Out` can be
```

**Pull request merged**